src/dom-to-image-more.js

```diff
@@ -257,24 +258,38 @@
         return Promise.resolve(node)
             .then(makeNodeCopy)
             .then(function (clone) {
-                return cloneChildren(node, clone);
+                return cloneChildren(util.hasOpenShadowRoot(node) ? node.shadowRoot : node, clone);
             })
             .then(function (clone) {
                 return processClone(node, clone);
             });
 
         function makeNodeCopy(original) {
-            return util.isHTMLCanvasElement(original)
-                ? util.makeImage(original.toDataURL())
-                : original.cloneNode(false);
+            if (util.isHTMLCanvasElement(original)) {
+                return util.makeImage(original.toDataURL());
+            }
+            if (util.isInShadowRoot(original)) {
+                return cloneShadowNode(original);
+            }
+            return original.cloneNode(false);
+        }
+
+        function cloneShadowNode(original) {
+            const nextSibling = original.nextSibling;
```

Comment:
Will `original.nextSibling` being null/nothing ever be an issue down on line 283?

Reply:
Don't worry about it - [Node.insertBefore() - Web APIs | MDN (mozilla.org)](https://developer.mozilla.org/en-US/docs/Web/API/Node/insertBefore#example_2)

Reply:
Line removed in d656297 regardless